### ISSUE TITLE
selectionStrategy not properly applied in specific scenario's

### ISSUE THREAD

**Issue by ThomasSwolfs** [SAP/cloud-sdk-js#4090]
**Describe the bug**
The library fails to fetch a destination, when selectionStrategy is imported from another location.

**To Reproduce**
Imagine following setup:

- A main application which uses the SAP Cloud SDK
- A custom library used by the main application. The custom library also uses SAP Cloud SDK

**Code in main application**
Main application performs a method imported from the custom library. The main app is passing through the information required to find the proper destination. But the custom library will be responsible for fetching the destination:

```
import { doSomething } from 'customlibrary1';
import { alwaysSubscriber } from '@sap-cloud-sdk/connectivity';

doSomething ({
                destinationName: 'MYDESTINATION',
                selectionStrategy: alwaysSubscriber,
                jwt: jwt
            }
```

**Custom library 1**
   

```
import {  getDestination } from '@sap-cloud-sdk/connectivity/dist/scp-cf';

export async function doSomething(destination: HttpDestinationOrFetchOptions) {
const destination = await getDestination({
        destinationName: destination.destinationName as string,
        selectionStrategy: destination.selectionStrategy,
        jwt: destination.jwt
    });
}
```
    

**Expected behavior**
The getDestination method used in the custom library, should return the destination for the subscriber account (based on selection strategy passed by the main application).

**Current behavior**
The isProviderNeeded and isSubscriberNeeded methods are comparing the selectionStrategy with the method defined inside of the package. However, in this scenario, the selectionStrategy passed through the methods is coming from the SAP Cloud SDK imported in the main application. So the reference to the object is not the same. This will result into the wrong selectionStrategy being applied.

**Screenshots**
![image](https://github.com/SAP/cloud-sdk-js/assets/9072117/3f5936f6-e95d-4e12-b01d-e1168b0a83f4)

**Comment by ThomasSwolfs** [comment]
I think that this can be solved by comparing the methods via a toString() method as pushed in this commit:

https://github.com/ThomasSwolfs/cloud-sdk-js/commit/8f5ebb1aaa3e415f7d78f42418d8976cbc5902da

**Comment by mr-flannery** [comment]
Hi @ThomasSwolfs ,

thanks for reporting the bug, I was able to reproduce it. 

Your fix looks good as well. Do you want to open a pull request or should I just open one myself?

Best regards,
Dennis

**Comment by ThomasSwolfs** [comment]
> 
@mr-flannery I created a pull request. However, I mistakenly committed with the wrong username.. So I'm stuck on signing the CLA as this username is not linked with a gitlab mail address.. :-)

https://github.com/SAP/cloud-sdk-js/pull/4094

**Comment by mr-flannery** [comment]
Addressed via #4095
